### PR TITLE
feat: phantomWallet support

### PR DESCRIPTION
**Pull request by thenoahhein**
Hey Rainbowkit team,

Opening this PR as we have a closed-beta going on right now, and several of the dapps that want to participate use rainbowkit for their modal.

Please let me know if there's anything I can do to support this PR and get it merged.

### CODE REVIEW

**Comment by chriskalani** [comment]
Hey @jeremygoldbrg 👋🏽 would love a review on this!

**Comment by mikearndt** [comment]
hey @chriskalani can you replace the rounded logo svg with a square one please?

**Comment by nheingit** [comment]
hey @mikearndt I've updated the logo — thanks for the quick response!

**Comment by nheingit** [comment]
Hey @mikearndt

Just wanted to follow up here to see if there were any blockers to getting this merged? Would love to let dapps downstream add Phantom to their modals.

**Comment by nickbytes** [comment]
@nheingit do you all use WalletConnect for you mobile wallet? If so, would you like to add it to the wallet connector here?

**Comment by nheingit** [comment]
@nickbytes We don't have walletconnect support out yet for ETH. We're currently integrated into WalletConnect on the solana side, and we're working with the WC team to see what it looks like for multichain. But nothing is live yet. We can write another PR once we've done the integration work on our end.

**Comment by nickbytes** [comment]
@nheingit would you mind adding Phantom to the example application in `/packages/example` ? It makes it easier to test changes w/ multiple wallets enabled.

File: packages/rainbowkit/src/wallets/walletConnectors/phantomWallet/phantomWallet.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { InjectedConnector } from 'wagmi/connectors/injected';
+import { Chain } from '../../../components/RainbowKitProvider/RainbowKitChainContext';
+import { Wallet } from '../../Wallet';
+
+export interface PhantomWalletOptions {
+  chains: Chain[];
+  shimDisconnect?: boolean;
+}
+
+export const phantomWallet = ({
```
Comment (nickbytes) [inline packages/rainbowkit/src/wallets/walletConnectors/phantomWallet/phantomWallet.ts:11]: We're missing the `installed` property here which detects whether the extension is installed or not. Check the Metamask wallet connector to see an example of how this is done @nheingit

Comment (nheingit) [reply packages/rainbowkit/src/wallets/walletConnectors/phantomWallet/phantomWallet.ts:11]: not super familiar with pnpm monorepos. To do this in the same way that the other extensions do, we'd need to bump wagmi to `0.9.1` which got cut yesterday. This updates the `ethereum` object in wagmi to have the `isPhantom` property needed to stay in line with the nice typings.

I can override it with an `any`, but wanted to get your opinion there.

Also curious what does `installed` actually do? In the most recent deployment I don't find any unexpected behavior.

**Comment by nickbytes** [comment]
@nheingit when using Phantom on the [example deployed link](https://rainbowkit-example-r3hmri7vx-rainbowdotme.vercel.app/), looks like it's not getting picked up as `installed`. Is the same happening for you?

**Comment by nickbytes** [comment]
Sorry @nheingit – I see your comment above about `9.1`. We can get this thing updated in another PR, all good.

**Comment by nheingit** [comment]
No @nickbytes it shows as installed for me, but that’s because I have the beta build installed. 

Your phantom won’t be injecting the provider object until you add a beta code to your build

**Comment by nheingit** [comment]
@nickbytes any movement here? Would love to get this merged in

**Comment by chriskalani** [comment]
@jeremygoldbrg @nickbytes can we get this over the finish-line this week please?

**Review by tian000 — APPROVED** [review]

File: examples/with-next/pages/_app.tsx [hunk 2]
```diff
@@ -10,6 +10,7 @@ import {
   argentWallet,
   trustWallet,
   ledgerWallet,
+  phantomWallet,
```
Comment (DanielSinclair) [inline examples/with-next/pages/_app.tsx:13]: Revert addition to `with-next`

File: site/components/Provider/Provider.tsx [hunk 3]
```diff
@@ -38,6 +39,7 @@ const connectors = connectorsForWallets([
       omniWallet({ chains }),
       imTokenWallet({ chains }),
       ledgerWallet({ chains }),
+      phantomWallet({ chains }),
```
Comment (DanielSinclair) [inline site/components/Provider/Provider.tsx:42]: Revert addition to `site`

**Review by DanielSinclair — CHANGES_REQUESTED** [review]
Please also migrate `DesktopOptions` and `ConnectDetails` changes to a new PR. We can stage them for the same release, but those will need additional auditing and testing. Likely can't proceed with this current implementation and would need more robust changes to `ready` upstream. The changes may need to be made upstream to Wagmi [here](https://github.com/wagmi-dev/references/blob/9c12708628c74b9ff677a50f5f6040083473e014/packages/connectors/src/injected.ts#L90)

**Review by nickbytes — APPROVED** [review]

**Review by DanielSinclair — APPROVED** [review]

**Pull request merged**